# Olist : Exploration et analyse des jeux de donnees

In [287]:
import pandas as pd 
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

Set option display max columns pour toujours afficher l'ensemble de nos colonnes et ne pas avoir de "...".

In [288]:
pd.set_option('display.max_columns', None)

## Creation des dataframes avec les datasets

In [289]:
#Data Frame qui va nous permettre de traduire nos autres dataframes en anglais (Portugais -> Anglais)
df_translation = pd.read_csv("csv/product_category_name_translation.csv")

#Data Frames respectifs des clients et des vendeurs
df_customers = pd.read_csv("csv/olist_customers_dataset.csv", dtype={'customer_zip_code_prefix' : str})
df_sellers = pd.read_csv("csv/olist_sellers_dataset.csv")

#Data Frame comportant des informations geographiques permettant d'etablir les distances entre nos vendeurs et nos acheteurs.
df_geolocation = pd.read_csv("csv/olist_geolocation_dataset.csv", dtype={'geolocation_zip_code_prefix' : str, 'geolocation_lat' : float, 'geolocation_lng' : float})

#Data Frame des produits vendus par Olist
df_products = pd.read_csv("csv/olist_products_dataset.csv")

#Data Frames relatifs aux commandes  
df_orders = pd.read_csv("csv/olist_orders_dataset.csv")
df_order_items = pd.read_csv("csv/olist_order_items_dataset.csv")
df_order_payments = pd.read_csv("csv/olist_order_payments_dataset.csv")
df_order_reviews = pd.read_csv("csv/olist_order_reviews_dataset.csv")

![DB olist](olist_datas.png)

## Apercu des dataframes :

In [290]:
#customers
df_customers.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

  customer_zip_code_prefix          customer_city customer_state  
0                    14409                 franca             SP  
1                    09790  sao bernardo do campo             SP  
2                    01151              sao paulo             SP  
3                    08775        mogi das cruzes             SP  
4                    13056               campinas             SP

Customers va contenir principalement la ville et l'Etat du client.

In [291]:
#sellers
df_sellers.head()

seller_id  seller_zip_code_prefix  \
0  3442f8959a84dea7ee197c632cb2df15                   13023   
1  d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2  ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3  c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4  51a04a8a6bdcb23deccc82b0b80742cf                   12914   

         seller_city seller_state  
0           campinas           SP  
1         mogi guacu           SP  
2     rio de janeiro           RJ  
3          sao paulo           SP  
4  braganca paulista           SP

Meme chose pour le data frame Sellers.

In [292]:
#geolocation
df_geolocation.head()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                       01037       -23.545621       -46.639292   
1                       01046       -23.546081       -46.644820   
2                       01046       -23.546129       -46.642951   
3                       01041       -23.544392       -46.639499   
4                       01035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP

Geolocation comprend la latitude et la longitude ainsi que la ville et l'etat.

In [293]:
#orders
df_orders.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [294]:
df_order_items.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

order_id : identifiant unique pour chaque commande

order_item_id: identifie le nombre d'item inclus dans la meme commande.

product_id : identifiant unique du produit

seller_id : identifiant du vendeur

shipping_limit_date : date limite d'envoi pour le vendeur

price : le prix du produit

In [295]:
df_order_payments.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

order_id: identifiant unique d'une commande

payment_sequential: un client peut payer avec plus d'un moyen de paiement. Pour chaque moyen de paiement une sequence est cree.

payment_type: methode de paiement du client

payment_installments: nombre de versements choisi par le client

payment_value: valeur de la transaction

In [296]:
df_order_reviews.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53

review_id: identifiant unique de la review

order_id: identifiant unique de la commande

review_score: score /5 de la review

review_comment_title: titre

review_comment_message: message 

review_creation_date: date d'envoi du rapport de satisfaction au client

review_answer_timestamp: date d'envoi du retour du client

## Modification du Data Frame "Review"

Dans ce projet, je ne vais pas traiter les commentaires rediges par les clients. Je fais le choix de supprimer les colonnes title et message, et de garder la colonne score.

In [297]:
df_order_reviews = df_order_reviews.drop('review_comment_title', axis=1)
df_order_reviews = df_order_reviews.drop('review_comment_message', axis=1)
df_order_reviews.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_creation_date review_answer_timestamp  
0             4  2018-01-18 00:00:00     2018-01-18 21:46:59  
1             5  2018-03-10 00:00:00     2018-03-11 03:05:13  
2             5  2018-02-17 00:00:00     2018-02-18 14:36:24  
3             5  2017-04-21 00:00:00     2017-04-21 22:02:06  
4             5  2018-03-01 00:00:00     2018-03-02 10:26:53

## df_orders

In [298]:
df_orders.head(1)

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00

In [299]:
df_orders.shape

(99441, 8)

In [300]:
df_order_items.head(1)

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   

   shipping_limit_date  price  freight_value  
0  2017-09-19 09:45:35   58.9          13.29

In [301]:
df_order_items.shape

(112650, 7)

In [302]:
df_order_payments.head(1)

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33

In [303]:
df_order_payments.shape

(103886, 5)

In [304]:
df_order_reviews.head(1)

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   

   review_score review_creation_date review_answer_timestamp  
0             4  2018-01-18 00:00:00     2018-01-18 21:46:59

In [305]:
df_order_reviews.shape

(99224, 5)

In [306]:
data = pd.merge(df_orders, df_order_items, on="order_id")
data = data.merge(df_order_payments, on="order_id")
data = data.merge(df_order_reviews, on="order_id")
data.head(1)


order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  payment_sequential payment_type  \
0  2017-10-06 11:07:15  29.99           8.72                   1  credit_card   

   payment_installments  payment_value                         review_id  \
0                     1          18.12  a54f0611adc9ed256b57ede6b6eb5114   

   review_score review_creation_date review_answer_timestamp  
0             4  2017-10-11 00:00:00     2017-10-12 03:43:48

In [307]:
data.shape

(117329, 22)

Je merge mes dataframes orders "on" order_id dans une dataframe "data".

## df_products

In [308]:
df_products.head(1)

product_id product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5            perfumaria   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0

In [309]:
df_products.shape

(32951, 9)

In [310]:
data = data.merge(df_products, on="product_id")

In [311]:
data.shape

(117329, 30)

## Formating des data frames customers, sellers et geolocation

Je vais dans un premier temps rename mes colonnes des 3 dataframes pour me permettre de merge la geolocation dans les dataframes customers et sellers.

In [312]:
df_geolocation = df_geolocation.rename(columns={"geolocation_zip_code_prefix" : "zip_code_prefix", "geolocation_city" : "city", "geolocation_state" : "state"})

In [313]:
df_geolocation.head(1)

zip_code_prefix  geolocation_lat  geolocation_lng       city state
0           01037       -23.545621       -46.639292  sao paulo    SP

In [314]:
df_customers = df_customers.rename(columns={"customer_zip_code_prefix" : "zip_code_prefix", "customer_city" : "city", "customer_state" : "state"})

In [315]:
df_customers.head(1)

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   

  zip_code_prefix    city state  
0           14409  franca    SP

In [316]:
df_sellers = df_sellers.rename(columns={"seller_zip_code_prefix" : "zip_code_prefix", "seller_city" : "city", "seller_state" : "state"})

In [317]:
df_sellers.head(1)

seller_id  zip_code_prefix      city state
0  3442f8959a84dea7ee197c632cb2df15            13023  campinas    SP

In [318]:
df_geolocation['zip_code_prefix'].value_counts().describe()

count    19015.000000
mean        52.598633
std         72.057907
min          1.000000
25%         10.000000
50%         29.000000
75%         66.500000
max       1146.000000
Name: zip_code_prefix, dtype: float64

Il y a 19015 differents zipcodes, un zipcode possede 1146 occurences, avec une moyenne d'occurence d'environ 53 valeurs par zipcode. 

In [319]:
df_geolocation['geolocation_lat'].dtype

dtype('float64')

In [320]:
df_geolocation['geolocation_lat'].describe()

count    1.000163e+06
mean    -2.117615e+01
std      5.715866e+00
min     -3.660537e+01
25%     -2.360355e+01
50%     -2.291938e+01
75%     -1.997962e+01
max      4.506593e+01
Name: geolocation_lat, dtype: float64

Le territoire territoire du bresil s'étend sur 4 395 km du Nord au Sud (5°16'20" Nord à 33°44'32" Sud de latitude) (5.27 | -33.74) et 4 319 km d'Est en Ouest (34°47'30" Est à 73°59'32" Ouest de longitude) (-33.79 | -73.99) et couvre trois fuseaux horaires. Je fais donc un traitement sur les donnees de la longitude et de la latitude afin de supprimer les valeurs incoherentes.

In [321]:
df_geolocation = df_geolocation[df_geolocation['geolocation_lat'] <= 5.27]
df_geolocation = df_geolocation[df_geolocation['geolocation_lat'] >= -33.74]
df_geolocation = df_geolocation[df_geolocation['geolocation_lng'] <= -33.79]
df_geolocation = df_geolocation[df_geolocation['geolocation_lng'] >= -73.99]

Je remarque que pour chaque zipcode il existe parfois des dizaines de latitudes et de longitudes differentes. Je veux avoir une valeur moyenne de latitude et de longitude pour chaque zipcode. Cependant je decide d'utiliser la mediane afin d'eviter que mes outliers pertubent fortement mes moyennes. 

In [322]:
df_geolocation[df_geolocation['zip_code_prefix'] == '14409']

zip_code_prefix  geolocation_lat  geolocation_lng    city state
334422           14409       -20.509897       -47.397866  franca    SP
334449           14409       -20.497396       -47.399241  franca    SP
334459           14409       -20.510459       -47.399553  franca    SP
334462           14409       -20.480940       -47.394161  franca    SP
334463           14409       -20.515413       -47.398194  franca    SP
...                ...              ...              ...     ...   ...
336625           14409       -20.483960       -47.403324  franca    SP
336650           14409       -20.483960       -47.403324  franca    SP
336653           14409       -20.487255       -47.405925  franca    SP
336657           14409       -20.510459       -47.399553  franca    SP
336670           14409       -20.500253       -47.400367  franca    SP

[147 rows x 5 columns]

In [323]:
df_geo = df_geolocation.groupby('zip_code_prefix')['geolocation_lat', 'geolocation_lng'].median().reset_index()
df_geo

/tmp/ipykernel_16377/3936784339.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df_geo = df_geolocation.groupby('zip_code_prefix')['geolocation_lat', 'geolocation_lng'].median().reset_index()


zip_code_prefix  geolocation_lat  geolocation_lng
0               01001       -23.550381       -46.634027
1               01002       -23.548551       -46.635072
2               01003       -23.548977       -46.635313
3               01004       -23.549535       -46.634771
4               01005       -23.549612       -46.636532
...               ...              ...              ...
19005           99960       -27.953797       -52.029641
19006           99965       -28.179542       -52.035551
19007           99970       -28.343273       -51.873734
19008           99980       -28.388092       -51.846880
19009           99990       -28.329595       -51.769362

[19010 rows x 3 columns]

Il y a 19010 zip code differents.

In [324]:
df_geolocation.groupby('zip_code_prefix')['city'].unique()

zip_code_prefix
01001    [sao paulo, são paulo]
01002    [sao paulo, são paulo]
01003    [sao paulo, são paulo]
01004    [sao paulo, são paulo]
01005    [sao paulo, são paulo]
                  ...          
99960                 [charrua]
99965              [agua santa]
99970        [ciriaco, ciríaco]
99980         [david canabarro]
99990               [muliterno]
Name: city, Length: 19010, dtype: object

Je vois ici que certains zip_code possedent "plusieurs villes"

In [325]:
df_geolocation.groupby('zip_code_prefix')['city'].value_counts()

zip_code_prefix  city           
01001            sao paulo          23
                 são paulo           3
01002            sao paulo          12
                 são paulo           1
01003            sao paulo          16
                                    ..
99965            agua santa          6
99970            ciriaco            20
                 ciríaco             1
99980            david canabarro    26
99990            muliterno           2
Name: city, Length: 27900, dtype: int64

Certaines villes ont plusieurs zipcode donc je decide de ne pas garder la ville et l'etat par la suite. Je garde le zipcode pour le moment uniquement pour merge mon dataframe geolocation. Mon dataframe df_sellers devient donc inutile dans mon dataframe data car il apporte uniquement le zipcode du seller qui ne me servira pas. Je pourrais eventuellement l'utiliser pour definir la distance entre mon seller et mon customer.

In [326]:
df_customers = df_customers.drop(['state', 'city'], axis=1)
df_customers.head(1)

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   

  zip_code_prefix  
0           14409

In [327]:
df_sellers = df_sellers.drop(['state', 'city'], axis=1)
df_sellers.head(1)

seller_id  zip_code_prefix
0  3442f8959a84dea7ee197c632cb2df15            13023

## Merge des derniers dataframes sur Data

Je merge customers sur data afin de recuperer les zipcodes pour pouvoir ajouter la latitude et la longitude de mes customers.

In [328]:
data.head(1)

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  payment_sequential payment_type  \
0  2017-10-06 11:07:15  29.99           8.72                   1  credit_card   

   payment_installments  payment_value                         review_id  \
0                     1          18.12  a54f0611adc9ed256b57ede6b6eb5114   

   review_score review_creation_date review_answer_timestamp  \
0             4  2017-10-11 00:00:00     2017-10-12 03:43:48   

   product_category_name  product_name_lenght  product_description_lenght  \
0  utilidades_domesticas                 40.0                       268.0   

   product_photos_qty  product_weight_g  product_length_cm  product_height_cm  \
0                 4.0             500.0               19.0                8.0   

   product_width_cm  
0              13.0

In [329]:
data.shape

(117329, 30)

In [330]:
df_customers.head(1)

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   

  zip_code_prefix  
0           14409

In [331]:
df_customers.dtypes

customer_id           object
customer_unique_id    object
zip_code_prefix       object
dtype: object

In [332]:
data = data.merge(df_customers, on="customer_id")
data.head(1)

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  payment_sequential payment_type  \
0  2017-10-06 11:07:15  29.99           8.72                   1  credit_card   

   payment_installments  payment_value                         review_id  \
0                     1          18.12  a54f0611adc9ed256b57ede6b6eb5114   

   review_score review_creation_date review_answer_timestamp  \
0             4  2017-10-11 00:00:00     2017-10-12 03:43:48   

   product_category_name  product_name_lenght  product_description_lenght  \
0  utilidades_domesticas                 40.0                       268.0   

   product_photos_qty  product_weight_g  product_length_cm  product_height_cm  \
0                 4.0             500.0               19.0                8.0   

   product_width_cm                customer_unique_id zip_code_prefix  
0              13.0  7c396fd4830fd04220f754e42b4e5bff           03149

Je merge ma geolocation sur data

In [333]:
df_geo.head(1)

zip_code_prefix  geolocation_lat  geolocation_lng
0           01001       -23.550381       -46.634027

In [334]:
data['zip_code_prefix']

0         03149
1         03149
2         03149
3         03366
4         02290
          ...  
117324    03318
117325    03318
117326    83870
117327    12209
117328    83750
Name: zip_code_prefix, Length: 117329, dtype: object

In [335]:
df_geo['zip_code_prefix']

0        01001
1        01002
2        01003
3        01004
4        01005
         ...  
19005    99960
19006    99965
19007    99970
19008    99980
19009    99990
Name: zip_code_prefix, Length: 19010, dtype: object

In [336]:
data = data.merge(df_geo, on="zip_code_prefix")
data.head(1)

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  payment_sequential payment_type  \
0  2017-10-06 11:07:15  29.99           8.72                   1  credit_card   

   payment_installments  payment_value                         review_id  \
0                     1          18.12  a54f0611adc9ed256b57ede6b6eb5114   

   review_score review_creation_date review_answer_timestamp  \
0             4  2017-10-11 00:00:00     2017-10-12 03:43:48   

   product_category_name  product_name_lenght  product_description_lenght  \
0  utilidades_domesticas                 40.0                       268.0   

   product_photos_qty  product_weight_g  product_length_cm  product_height_cm  \
0                 4.0             500.0               19.0                8.0   

   product_width_cm                customer_unique_id zip_code_prefix  \
0              13.0  7c396fd4830fd04220f754e42b4e5bff           03149   

   geolocation_lat  geolocation_lng  
0        -23.57617       -46.587276

Maintenant que j'ai ma latitude et et ma longitude dans data je peux supprimer zip_code pour eviter la redondance d'information.

In [337]:
data = data.drop('zip_code_prefix', axis=1)
data.head(1)

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   

                         product_id                         seller_id  \
0  87285b34884572647811a353c7ac498a  3504c0cb71d7fa48d967e0e4c94d59d9   

   shipping_limit_date  price  freight_value  payment_sequential payment_type  \
0  2017-10-06 11:07:15  29.99           8.72                   1  credit_card   

   payment_installments  payment_value                         review_id  \
0                     1          18.12  a54f0611adc9ed256b57ede6b6eb5114   

   review_score review_creation_date review_answer_timestamp  \
0             4  2017-10-11 00:00:00     2017-10-12 03:43:48   

   product_category_name  product_name_lenght  product_description_lenght  \
0  utilidades_domesticas                 40.0                       268.0   

   product_photos_qty  product_weight_g  product_length_cm  product_height_cm  \
0                 4.0             500.0               19.0                8.0   

   product_width_cm                customer_unique_id  geolocation_lat  \
0              13.0  7c396fd4830fd04220f754e42b4e5bff        -23.57617   

   geolocation_lng  
0       -46.587276

In [338]:
data.shape

(117012, 33)